http_client/models/http_client_auth.py

```diff
@@ -268,6 +272,9 @@ def get_data(self):
         auth_method = self.auth_method
         if auth_method == 'basic':
             data['auth'] = {'auth': (self.identifier, self.secret)}
+        elif auth_method == 'api_key':
+            header_key = self.identifier or DEFAULT_API_KEY_HEADER
+            data['auth'] = {'headers': {header_key: self.secret}}
```

Comment:
We should not fallback to `DEFAULT_API_KEY_HEADER`, because then it would be possible to have empty identifier and silently use default without user realizing it. Instead we have to make `identifier` required on view, to indicate that it needs to be specified when `api_key` method is used.
```suggestion
            data['auth'] = {'headers': {self.identifier: self.secret}}
```